Tests/PackageGraphTests/PackageGraphTests.swift

```diff
@@ -818,6 +898,49 @@ class PackageGraphTests: XCTestCase {
         }
     }
 
+    func testInvalidExplicitPackageDependencyName() {
+        let fs = InMemoryFileSystem(emptyFiles:
+            "/Foo/Sources/Foo/foo.swift",
+            "/Bar/Sources/Baar/bar.swift"
+        )
+
+        let diagnostics = DiagnosticsEngine()
+        _ = loadPackageGraph(fs: fs, diagnostics: diagnostics,
+            manifests: [
+                Manifest.createV4Manifest(
+                    name: "Foo",
+                    path: "/Foo",
+                    url: "/Foo",
+                    packageKind: .root,
+                    dependencies: [
+                        PackageDependencyDescription(name: "Baar", url: "/Bar", requirement: .upToNextMajor(from: "1.0.0")),
+                    ],
+                    targets: [
+                        TargetDescription(name: "Foo", dependencies: ["Baar"]),
+                    ]),
+                Manifest.createV4Manifest(
+                    name: "Bar",
+                    path: "/Bar",
+                    url: "/Bar",
+                    packageKind: .local,
+                    products: [
+                        ProductDescription(name: "Baar", targets: ["Baar"])
+                    ],
+                    targets: [
+                        TargetDescription(name: "Baar"),
+                    ]),
+            ]
+        )
+
+        DiagnosticsEngineTester(diagnostics, ignoreNotes: true) { result in
+            result.check(
+                diagnostic: "package dependency 'Baar' does not match the package name 'Bar'",
```

Comment:
How about something like:

"The name declared ('Baar') for dependency `"/Bar"` doesn't match the actual package name `'Bar'`.

Reply:
Improved it, thanks!